frontend/js/behaviors/layeredImageViewer/layeredImageViewer.js

```diff
@@ -185,19 +237,68 @@ import OpenSeadragon from '../../libs/openseadragon4.min';
    * @method
    */
   _initViewer() {
-    // OSD expects no siblings, by having a mount element
+    // OSD expects no siblings, keep it an orphan
     // the destroy() method will preserve the initial HTML
-    const mountEl = document.createElement('div');
-    this.mountEl = mountEl;
-    this.mountEl.classList.add('o-layered-image-viewer__osd-mount');
-    this.element.insertAdjacentElement('beforeend', mountEl);
 
-    this.mountEl.style.display = 'block';
+    const mediaTemplate = document.createElement('template');
+    let captionMarkup = '';
+
+    // Markup to be inserted if captionTitle / caption exist
+    if (this.captionEl || this.captionTitleEl) {
+      captionMarkup = `<figcaption>`;
+      if (this.captionTitleEl) {
+        captionMarkup += `
+          <div class="f-caption-title">
+            ${this.captionTitleEl.innerHTML}
+          </div>`;
+      }
+      if (this.captionEl) {
+        captionMarkup += `
+          <div class="f-caption">
+            ${this.captionEl.innerHTML}
+          </div>`;
+      }
+      captionMarkup += `</figcaption>`;
+    }
+
+    // Markup to surround viewer with
+    // Modelled around the media molecule
+    mediaTemplate.innerHTML = `
+      <figure class="m-media m-media--${this.size} m-media--contain o-blocks__block">
+        <div class="m-media__img m-media--layered-image-viewer-embed">
+          <div class="o-layered-image-viewer__osd-mount"></div>
+        </div>
+        ${captionMarkup}
+      </figure>
```

Comment:
All the HTML inside of the JS feels messy. Does all this HTML have to live here? Could the HTML exist in the Blade files and classes get added/removed to affect the FE behavior?

Reply:
Thanks for the context, Jon. If this has been tricky to get all the pieces to coordinate, I don't want to over-complicate how you've assembled things. We generally prefer to minimize how much HTML is constructed in our JS. Having shared that, we'll defer to you all on the cleanest way to organize the code you're building.